libminifi/include/utils/ProcessorConfigUtils.h

```diff
@@ -30,7 +30,15 @@ namespace nifi {
 namespace minifi {
 namespace utils {
 
-std::string getRequiredPropertyOrThrow(const core::ProcessContext* context, const std::string& property_name);
+template<typename PropertyType = std::string>
+PropertyType getRequiredPropertyOrThrow(const core::ProcessContext* context, const std::string& property_name) {
```

Comment:
Can we make the `context` parameter a reference instead of a pointer? It's UB to call this with null, so we might as well reflect it in the signature itself.

Reply:
Changed in 98ed8e2450436c2502a71f0bc99301fa4576db2c